### ISSUE TITLE
[Enhancement] eventmesh-runtime whether to consider using spring or spring boot or guice framework to improve development efficiency

### ISSUE THREAD

**Issue by weihubeats** [apache/incubator-eventmesh#2314]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

eventmesh-runtime whether to consider using spring or spring boot or guice framework to improve development efficiency and code readability

### Describe the solution you'd like

use spring or spring boot or guice framework to improve development efficiency and code readability

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

**Comment by horoc** [comment]
I think maybe it's a good idea to integrate with spring in Java SDK which can make EventMesh more friendly to Java users, but about the runtime module, In my point of view, I prefer to make the core module independent and decoupled from other frameworks like spring. Maybe we can start a discussion first.

**Comment by weihubeats** [comment]
Is the core module independent to be more lightweight? If you think spring is too heavy, can you consider using [guice](https://github.com/google/guice) or do you think the core module is supposed to reduce three-way dependencies

**Comment by weihubeats** [comment]
Java SDK is to provide spring boot sdk, or directly on the basis of the original transformation it

**Comment by horoc** [comment]
> Is the core module independent to be more lightweight? If you think spring is too heavy, can you consider using [guice](https://github.com/google/guice) or do you think the core module is supposed to reduce three-way dependencies

right, I prefer to reduce extra dependencies if it doesn't refer to some functionality feature. 

What do you think? @xwm1992

**Comment by horoc** [comment]
> Java SDK is to provide spring boot sdk, or directly on the basis of the original transformation it

I mean, currently, before Java users request to EventMesh server,  they should set up the client manually, maybe we can provide Spring annotations to simplify the process.

<img width="858" alt="image" src="https://user-images.githubusercontent.com/63457588/204788323-875c56cf-dd27-4d30-aec5-ec550d755c4c.png">

**Comment by weihubeats** [comment]
> > Java SDK is to provide spring boot sdk, or directly on the basis of the original transformation it
> 
> I mean, currently, before Java users request to EventMesh server, they should set up the client manually, maybe we can provide Spring annotations to simplify the process.
> 
> <img alt="image" width="858" src="https://user-images.githubusercontent.com/63457588/204788323-875c56cf-dd27-4d30-aec5-ec550d755c4c.png">

Yes, if you use spring boot it will simplify a lot of configuration and code writing

**Comment by HattoriHenzo** [comment]
I like the idea of using Spring but I think it should only be suitable to create a SDK for Spring only. Java is not only about Spring and it can be good to keep the Java SDK as it is (as a king of vanilla Java)

**Comment by mroccyen** [comment]
What is the current progress? I could try it.

**Comment by huyuanxin** [comment]
IMO, it's a good idea to provide Spring annotations, but we need to keep the java SDK.

Developers who using spring or spring boot can define the client easily, and the others still can use the SDK.

So, a dependence for spring or a starter for sping-boot is needed.
